### PR TITLE
add formatting check only on changed lines

### PR DESCRIPTION
**Pull request by sbera87**
*Issue #, if available:*
V1506947992

*Description of changes:*

*Check all that applies:*
- [ x] Did a review by yourself.
- [ ] Added proper tests to cover this PR. (If tests are not applicable, explain.)
- [ ] Checked if this PR is a breaking (APIs have been changed) change.
- [ ] Checked if this PR will _not_ introduce cross-platform inconsistent behavior.
- [ ] Checked if this PR would require a ReadMe/Wiki update.

Check which platforms you have built SDK on to verify the correctness of this PR.
- [ ] Linux
- [ ] Windows
- [ ] Android
- [ ] MacOS
- [ ] IOS
- [ ] Other Platforms

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/clang-format.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: clang-format-check
```
Comment (sbera87) [inline .github/workflows/clang-format.yml:1]: Tested this by creating new repo and using this action with llvm style flagged to be formatted into Google style.
Verified works only with the differences in a file and not entire file

Comment (sbera87) [reply .github/workflows/clang-format.yml:1]: Test Example:
Added ill-formatted cpp code
Screenshot 2024-11-01 at 11 46 15 AM

Comment (sbera87) [reply .github/workflows/clang-format.yml:1]: Github actions catch formatting issues as expected on the PR

Screenshot 2024-11-01 at 11 46 46 AM

**Comment by sbera87** [comment]
To run formatter on your changes
git diff -U0 --no-color main <ur branch> | python3 clang-format-diff.py -p1 -i

**Review by SergeyRyabinin — APPROVED** [review]

**Pull request merged**